src/client/app/comm.ts

```diff
@@ -1,17 +1,25 @@
+import { BaseModel } from '../model/BaseModel';
+import { Content } from '../model/Content';
 import { Problem } from '../model/Problem';
 import { ProblemResponse } from '../model/ProblemResponse';
 
+export const CONTENT_CLASS: Content = new Content().constructor as any;
+export const PROBLEM_CLASS: Problem = new Problem().constructor as any;
+
 function getUserId(): string {
     const inputField: HTMLElement = document.getElementById('hmhUserId');
     return (inputField as HTMLInputElement).value || '12345';
 }
 
-export async function loadProblem(id: string): Promise<Problem> {
-    return fetch('/api/v1/ang-eng/Problem/' + id, { headers: { accept: 'application/json', 'X-HMH-User-Id': getUserId() }, method: 'GET' })
+export async function loadEntity(id: string, modelClass: BaseModel): Promise<BaseModel> {
+    return fetch('/api/v1/ang-eng/' + modelClass.name + '/' + id, {
+        headers: { accept: 'application/json', 'X-HMH-User-Id': getUserId() },
+        method: 'GET'
+    })
         .then((response: Response): any => response.json())
         .then(
             (model: any): Problem => {
-                return new Problem().fromHttp(model);
+                return modelClass.getInstance().fromHttp(model);
```

Comment:
Why can't you do `ContentItem.getInstance()`

Reply:
Because now I want to offer the same function to any entity of class {`Content`, `Problem`, `Image`, etc.}. By letting callers of the function specifying which class to process, the URL is built automatically and the JSON payload is converted to the expected class.

Note that I've introduced the same type of functions for the `bootstrap` code (see src/server/scripts/comm.js) and let me know if I should rename this `loadEntity` in `httpGet` too here ;)

Reply:
`httpGet` seems a better name than `loadEntity`.

Reply:
Now named `httpGet` and `httpCreate` ;)